dissect/target/plugins/os/windows/catroot.py

```diff
@@ -60,67 +114,135 @@ def catroot(self):
+
+                    file_digest = digest()
+                    if len(hexdigest) == 32:
+                        file_digest.md5 = hexdigest
+                    elif len(hexdigest) == 40:
+                        file_digest.sha1 = hexdigest
+                    elif len(hexdigest) == 64:
+                        file_digest.sha256 = hexdigest
+
+                    digests.append(file_digest)
+
+                # Finding the hint in encap_content_info is on best effort. In most of the cases,
+                # there is a key "PackageName" available. We first try to parse the corresponding
+                # value if it is present. If this does not succeed, we might be dealing with catroot
+                # files containing the "hint" needle.
+                # If both methods do not result in a file hint, there is either no hint available or
+                # the format is not yet known and therefore not supported.
+                hints = []
+                try:
+                    if offset:
+                        # As far as known, the PackageName data is only present in the encap_content_info
+                        # after the last digest.
+                        hint_buf = encap_contents[offset + len(needle) + len(raw_digest) + 2 :]
+
+                        # First try to find to find the "PackageName" value, if it's present.
+                        hint = find_package_name(hint_buf)
+                        if hint:
+                            hints.append(hint)
+
+                    # If the package_name needle is not found or it's not present in the first 7 bytes of the hint_buf
+                    # We are probably dealing with a catroot file that contains "hint" needles.
+                    if not hints:
+                        for hint_offset in findall(encap_contents, NEEDLES["hint"]):
+                            # Either 3 or 4 bytes before the needle, a sequence starts
+                            bytes_before_needle = 3 if encap_contents[hint_offset - 3] == 48 else 4
+                            name_sequence = Sequence.load(encap_contents[hint_offset - bytes_before_needle :])
+
+                            hint = name_sequence[2].native.decode("utf-16-le").strip("\x00")
+                            hints.append(hint)
+
+                except Exception as error:
+                    self.target.log.warning(
+                        "An error occurred while parsing the hint for catroot file %s: %s", file, error
+                    )
+
+                # Currently, it is not known how the file hints are related to the digests. Therefore, each digest
+                # is yielded as a record with all of the file hints found.
+                for file_digest in digests:
+                    yield CatrootRecord(
+                        digest=file_digest,
+                        hints=hints,
+                        filename=file.name,
+                        source=self.target.fs.path(file),
```

Comment:
```suggestion
                        source=file,
```
It's already of the correct type.